src/Razor/test/Microsoft.VisualStudio.LanguageServices.Razor.Test/Cohost/CohostOnAutoInsertEndpointTest.cs

```diff
@@ -0,0 +1,136 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.Threading.Tasks;
+using Microsoft.CodBeAnalysis.Remote.Razor.AutoInsert;
+using Microsoft.CodeAnalysis.ExternalAccess.Razor;
+using Microsoft.CodeAnalysis.Razor.AutoInsert;
+using Microsoft.CodeAnalysis.Razor.Settings;
+using Microsoft.CodeAnalysis.Testing;
+using Microsoft.CodeAnalysis.Text;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+using Microsoft.VisualStudio.LanguageServices.Razor.LanguageClient.Cohost;
+using Microsoft.VisualStudio.Razor.Settings;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.VisualStudio.Razor.LanguageClient.Cohost;
+
+public class CohostOnAutoInsertEndpointTest(ITestOutputHelper testOutputHelper) : CohostEndpointTestBase(testOutputHelper)
+{
+    [Theory]
+    [InlineData("PageTitle", "$0</PageTitle>", ">")]
+    [InlineData("div", "$0</div>", ">")]
+    [InlineData("text", "$0</text>", ">")]
+
+    public async Task Component_AutoInsertEndTag(string startTag, string endTag, string triggerCharacter)
+    {
+        var input = $"""
+            This is a Razor document.
+
+            <{startTag}$$
+
+            The end.
+            """;
+
+        await VerifyOnAutoInsertAsync(input, endTag, triggerCharacter);
+    }
+
+    [Theory]
+    [InlineData("div style", "\"\"", "=")]
```

Comment:
A theory with one case is a fact 😛

Reply:
I will move the data to test code here per your suggestion (I think we only have this one case HTML handles, right?)